#! /bin/sh /usr/share/dpatch/dpatch-run
## 04_fix-lucene-lock-directory.dpatch by Jan-Pascal van Best <janpascal@vanbest.org>
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: Fix FilePermission security exceptions because the default lucene
## DP: lock dir is the system temp dir, which under Tomcat is
## DP: /var/lib/tomcat5.5/temp. Move it to the dataDir (/var/lib/solr/data)
## DP: instead.

@DPATCH@
diff -urNad trunk~/src/java/org/apache/solr/core/SolrCore.java trunk/src/java/org/apache/solr/core/SolrCore.java
--- trunk~/src/java/org/apache/solr/core/SolrCore.java	2006-12-16 22:34:15.000000000 +0100
+++ trunk/src/java/org/apache/solr/core/SolrCore.java	2007-06-01 11:45:34.000000000 +0200
@@ -125,6 +125,7 @@
 
   void initIndex() {
     try {
+      System.setProperty( "org.apache.lucene.lockDir", dataDir );
       File dirFile = new File(getIndexDir());
       boolean indexExists = dirFile.canRead();
 
